cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/AbstractConstraint.java

```diff
@@ -24,13 +24,15 @@
 /**
  * A constraint that is checked before executing a schedule.
  */
-public class ConstraintChecker {
+public abstract class AbstractConstraint implements Constraint {
```

Comment:
Is something like this what you're recommending?
`DelayConstraint` ==> `AbstractConstraint` ==> `JobConstraint` ==> `Constraint`

`DelayConstraint` is a concrete class in cdap-app-fabric.
`AbstractConstraint` is an abstract class in cdap-app-fabric.
`JobConstraint` is an interface in cdap-app-fabric.
`Constraint` is an interface in cdap-api.